src/curses/ui_call_flow.c

```diff
@@ -641,37 +641,19 @@ call_flow_draw_rtp_stream(ui_t *ui, call_flow_arrow_t *arrow, int cline)
     callid = stream->media->msg->call->callid;
     msg_src = stream->media->msg->packet->src;
     msg_dst = stream->media->msg->packet->dst;
-    stream_src = stream->src;
-    stream_src.port = 0;
-    stream_dst = stream->dst;
-    stream_dst.port = 0;
-
-    // Get origin column for this stream.
-    // If we share the same Address from its setup SIP packet, use that column instead.
-    if (!strcmp(stream->src.ip, msg_src.ip)) {
-        column1 = call_flow_column_get(ui, callid, msg_src);
-    } else if (!strcmp(stream->src.ip, msg_dst.ip)) {
-        column1 = call_flow_column_get(ui, callid, msg_dst);
-    } else {
-        column1 = call_flow_column_get(ui, 0, stream_src);
-    }
 
-    // Get destination column for this stream.
-    // If we share the same Address from its setup SIP packet, use that column instead.
-    if (!strcmp(stream->dst.ip, msg_dst.ip)) {
-        column2 = call_flow_column_get(ui, callid, msg_dst);
-    } else if (!strcmp(stream->dst.ip, msg_src.ip)) {
-        column2 = call_flow_column_get(ui, callid, msg_src);
-    } else {
-        column2 = call_flow_column_get(ui, 0, stream_dst);
-    }
+    // Get call inititator source and destination address
+    src = stream->media->msg->call->cstart_msg->packet->src;
+    dst = stream->media->msg->call->cstart_msg->packet->dst;
```

Comment:
Sadly we can not make that kind of assumption, because between src and dst endpoints may be proxies that, within the same dialog, change the SDP at will.

I think this PR have two different changes: Improve SDP payload parsing to create rtp streams and fix rtp arrow representation.

For the second one, I have been debugging with the test file you provided and improve the column selection logic of rtp streams.
  * Now all arrows store their source and destination columns (also column1 and column2 were ambigous). 
* The direction of the arrow is determined by the displayed order of the source and destination columns.
 * Now RTP streams will try to find the SIP msg that setups the stream and use, if possible, the same columns that those messages. This will be done for both IP source and destination of the stream.

If you have time, give it a try to cf_rtp_arrows: It only changes visualization issues on call flow: the improvements of the SDP parsing logic have not been applied!

Thanks A LOT for your time and help!!